compiler/lib/parse_bytecode.ml

```diff
@@ -764,7 +767,24 @@ let access_global g i =
       x
 
 let register_global ?(force = false) g i loc rem =
-  if force || g.is_exported.(i)
+  if g.is_exported.(i)
+     &&
+     match Config.target () with
+     | `Wasm -> true
+     | `JavaScript -> false
+  then (
+    let name =
+      match g.named_value.(i) with
+      | None -> assert false
+      | Some name -> name
+    in
+    Code.Var.name (access_global g i) name;
+    ( Let
+        ( Var.fresh ()
+        , Prim (Extern "caml_set_global", [ Pc (String name); Pv (access_global g i) ]) )
```

Comment:
Can we try to unify the use of `caml_set_global` and `caml_register_global` ?

Reply:
With `wasm_of_ocaml`, `caml_set_global` set a global variable in the generated code, while `caml_register_global` register a value into the runtime system (this is used for exceptions).
I may reconsider this if we try to support dynamic linking, but for now they are doing very different things.

Reply:
Ok, I've pushed some small refactoring, can you check I didn't change the expected semantic ?

Reply:
Force is used to select which branch to follow.

Reply:
If the backend is JS and this branch is taken, `force` can have any value without changing the result.

Reply:
When registering a predefined exception, we set `force` to true so that this branch is taken.

Reply:
To make sure we read the same thing, here is the branch ```| true, _, (`JavaScript as target) | false, true, ((`Wasm | `JavaScript) as target) ->```

The branch is taken if exported or force is true. If force is true, it means we're registering an exception